src/main.rs

```diff
@@ -596,7 +596,8 @@ impl VideoPlayerInner {
                             if let Some(title) = info.get_title() {
                                 window.set_title(&*title);
                             } else {
-                                window.set_title(&*info.get_uri());
+                                let (filename, _) = glib::filename_from_uri(&uri).unwrap();
```

Comment:
Why not reusing the `info.get_uri()` value? Also the unwrap could be avoided I think. If the `filename_from_uri` call returns None, call the old code, something like

```
let info_uri = info.get_uri();
if let Some((filename, _)) =  glib::filename_from_uri(&uri) {
    window.set_title(&filename.as_os_str().to_string_lossy());
} else {
    window.set_title(&*info_uri);
}
```

Reply:
Thanks for suggestion @philn 
Totally agree, unwrap should be avoided. I've just modified the PR.